src/cpp/session/SessionSourceDatabaseSupervisor.cpp

```diff
@@ -116,8 +116,21 @@ FilePath sessionLockFilePath(const FilePath& sessionDir)
    return sessionDir.complete("lock_file");
 }
 
-// session dir lock (initialized by attachToSourceDatabase)
-FileLock s_sessionDirLock;
+// session dir lock (lock is acquired within 'attachToSourceDatabase()')
+boost::shared_ptr<FileLock> createSessionDirLock()
+{
+#ifdef RSTUDIO_SERVER
```

Comment:
I think this logic isn't quite what we want. Let's make the default type link base (which we know is safe for NFS) and then explicitly opt into Advisory when we are on the desktop.